### DISCUSSION TITLE
Example Config for agent with an external Srever?

### DISCUSSION THREAD

**Discussion by orsenthil** [woodpecker-ci/woodpecker#2628] [Ask a Question]
### Component

agent

### Describe the bug

I have a server and agent running on different machines.

I don't find any examples of agent only configuration that I can point to the woodpecker server.

I have tried this.

```
# docker-compose.yml
version: '3'

services:
  woodpecker-agent:
    image: woodpeckerci/woodpecker-agent:v0.15.4
    command: agent
    restart: always
    environment:
      - WOODPECKER_SERVER=ci.server.com:9000
      - WOODPECKER_AGENT_SECRET="SECRET"
      - WOODPECKER_DEBUG=true
      - WOODPECKER_BACKEND=docker
```

I get two different kinds of errors.

```
woodpecker-agent_1  | {"level":"error","error":"rpc error: code = Unavailable desc = connection closed before server preface received","time":"2023-10-21T19:20:12Z","message":"grpc error: done(): code: Unavailable: rpc error: code = Unavailable desc = connection closed before server preface received"}
```

and 

```
woodpecker-agent_1  | {"level":"error","error":"rpc error: code = Unavailable desc = connection closed before server preface received","time":"2023-10-23T00:51:47Z","message":"grpc error: done(): code: Unavailable: rpc error: code = Unavailable desc = connection closed before server preface received"}
```

### System Info

```shell
source	"https://github.com/woodpecker-ci/woodpecker"
version	"1.0.3"
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Checked that the bug isn't fixed in the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy) or the [Matrix room](https://matrix.to/#/#woodpecker:matrix.org).

**Comment by qwerty287** [comment]
Since I didn't try something like this, I can only guess what you could do:
- use IP address instead of domain in config
- there's an example in the docs which explicitly mentions agents connecting over the internet, but it's about reverse proxying (scroll down in https://woodpecker-ci.org/docs/administration/setup#docker-compose)

**Comment by anbraten** [comment]
Make sure you use v1.0 or next, but most importantly the same version for agent and server. Also you don't need `command: agent` and `WOODPECKER_BACKEND=docker`, but you are missing to mount the docker socket to your agent

**Comment by qwerty287** [comment]
Do you still face this issue?

**Comment by qwerty287** [comment]
Closing. Please comment if you still have this issue.
